src/strawberry_sqlalchemy_mapper/mapper.py

```diff
@@ -560,128 +574,158 @@ class Employee:
         ```
         """
 
-        def convert(type_: Any) -> Any:
-            old_annotations = getattr(type_, "__annotations__", {})
-            type_.__annotations__ = {}
-            mapper: Mapper = inspect(model)
-            generated_field_keys = []
-
-            excluded_keys = getattr(type_, "__exclude__", [])
-
-            # if the type inherits from another mapped type, then it may have
-            # generated resolvers. These will be treated by dataclasses as having
-            # a default value, which will likely cause issues because of keys
-            # that don't have default values. To fix this, we wrap them in
-            # `strawberry.field()` (like when they were originally made), so
-            # dataclasses will ignore them.
-            # TODO: Potentially raise/fix this issue upstream
-            for key in dir(type_):
-                val = getattr(type_, key)
-                if getattr(val, _IS_GENERATED_RESOLVER_KEY, False):
-                    setattr(type_, key, strawberry.field(resolver=val))
-                    generated_field_keys.append(key)
-
-            self._handle_columns(mapper, type_, excluded_keys, generated_field_keys)
-            for key, relationship in mapper.relationships.items():
-                relationship: RelationshipProperty
-                if (
-                    key in excluded_keys
-                    or key in type_.__annotations__
-                    or hasattr(type_, key)
-                ):
-                    continue
-                strawberry_type = self._convert_relationship_to_strawberry_type(
-                    relationship
-                )
-                self._add_annotation(
-                    type_,
-                    key,
+        def do_conversion(type_):
+            return self.convert(
+                type_,
+                model,
+                make_interface,
+                use_federation,
+            )
+
+        return do_conversion
+
+    def convert(
+        self,
+        type_: Any,
+        model: Type[BaseModelType],
+        make_interface=False,
+        use_federation=False,
+    ) -> Any:
+        """
+        Do type conversion. Usually accessed using typical .type decorator. But
```

Comment:
I think if this is to be part of the public API, we should probably beef up the docstring a bit. I still don't really understand when someone would need this, and why someone wouldn't just call `Employee = mapper.type(models.Employee)(Employee)`

At the least, exercising this in a unit test to have some basic API-level test coverage and demonstrate the usage could be nice, but might be worth waiting until Erik or Tim weighs in.